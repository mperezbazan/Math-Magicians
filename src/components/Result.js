import React from 'react';
import PropTypes from 'prop-types';

class Result extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const { value, columns } = this.props;
    return (
      <div className={`result dark-grey col-${columns}`}>
        {value}
      </div>
    );
  }
}
Result.propTypes = {
  value: PropTypes.number.isRequired,
  columns: PropTypes.number,
};
Result.defaultProps = {
  columns: 4,
};

export default Result;
