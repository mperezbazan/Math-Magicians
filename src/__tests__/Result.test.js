import renderer from 'react-test-renderer';
import Result from '../components/Result';

test('Match Result Snapshot', () => {
  const tree = renderer.create(<Result value="5" />).toJSON();
  expect(tree).toMatchSnapshot();
});
