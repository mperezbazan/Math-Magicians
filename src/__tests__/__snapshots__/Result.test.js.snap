// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match Result Snapshot 1`] = `
<div
  className="result dark-grey col-4"
>
  5
</div>
`;
